open-api.yaml

```diff
@@ -0,0 +1,262 @@
+        example: 'urn:de.fodjan.animal_data:69d4a4ec-e8a4-4768-a299-14a7a9acd1b1'
+    RationFeedstuffRequest:
+      title: RationFeedstuffRequest
+      type: object
+      properties:
+        amount_freshmatter:
+          type: number
+          description: Fresh matter amount of feedstuff in ration (kg)
+          format: double
+        cost_kg_freshmatter:
+          type: number
+          description: Fresh matter cost per kg of ration feedstuff (USD)
+          format: double
+          example: 0.4125
+        drymatter_percentage:
+          type: number
+          description: DM percentage of ration feedstuff
+          format: double
+          example: 87
+        feedstuff_id:
+          type: string
+          description: feedstuff_id represent identification number of ration feedstuff between systems
+          example: 49c78036-acfb-11ed-afa1-0242ac120002
+        name:
+          type: string
+          description: Name of ration feedstuff
+          example: Sojabohnen (gesch., Bioprofin)
+        type:
+          type: string
+          description: Feed type of feedstuff in ration, todo check completeness
+          enum: 
+            - dry_forage
+            - corn_silage
+            - alfa_silage
+            - grass_silage
+            - other_wet_forage
+            - by_product
+            - grain
+            - fatty_acid_source
+            - triglycerides
+            - cp_source
+            - rup_source
+            - mineral
+            - miscellaneous
+            - vitamin
+            - premix
+            - forage
+            - concentrate
+            - commercial
+            - water
+          example: dry_forage
+      required:
+        - amount_drymatter
+        - drymatter_percentage
+        - feedstuff_id
+        - name
+        - type
+    RationRequest:
+      title: RationRequest
+      type: object
+      properties:
+        ration_ref:
+          type: string
+          description: client-supplied reference for that ration. If ration_ref is not provided, new ration will be created. If ration_ref is already known, existing ration will be updated.
+          example: 62559890-acfb-11ed-afa1-0242ac120002
+        feedstuffs:
+          type: array
+          description: Order of feedstuff in this array will be used as order of feedstuff in mixer wagon
+          items:
+            $ref: '#/components/schemas/RationFeedstuffRequest'
+        pen_allocations:
+          type: array
+          items:
+            $ref: '#/components/schemas/PenAllocation'
+        name:
+          type: string
+          description: Ration name
+          example: Ration Test 1
+        start_date: 
+          type: string
+          format: date
+          description: Start date of ration, as ISO8601 date
+      required:
+        - feedstuffs
+        - pens
+        - name
+    PenAllocation:

```

Comment:
Please indicate which properties are required

Reply:
I have updated RationFeedstuffRequest and RationRequest models, the rest are response models

Reply:
It's good to know for implementors (and for consumers) which properties are required also in the response models

Reply:
indeed! I think most of the values should be required in the response.

Reply:
```suggestion
    PenAllocation:
      required: 
        - feed_intake_per_animal
        - urn
```